### PR TITLE
feat: optional ui

### PR DESCRIPTION
**Pull request by wesleytodd**
Move ui packages to dev deps and add handling for making them optional.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,18 +24,18 @@
     "express": "^4.18.2",
     "mocha": "^10.2.0",
     "standard": "^17.1.0",
-    "supertest": "^6.3.3"
+    "supertest": "^6.3.3",
+    "redoc": "^2.1.1",
```
Comment (enov) [inline package.json:28]: "redoc" and "swagger-ui-dist" should probably go under [optionalDependencies](https://docs.npmjs.com/cli/v6/configuring-npm/package-json?v=true#optionaldependencies), as they are not development dependencies.

**Review by enov — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (wesleytodd) [reply package.json:28]: IIRC the issue reported around redoc was at install time. I forget now, but I remember doing this on purpose. Happy to have you test it out and tell me otherwise, but my thinking here was it was more like a peer dep before npm actually started installing those. If you want it, you need to also install it in your app.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x, 18.x, 20.x]
+        node-version: [14.x, 16.x, 18.x, 20.x]
```
Comment (wesleytodd) [inline .github/workflows/test.yml:8]: @enov Ah, I see I did the same thing here. Sorry for correcting that, I had pushed this in a second commit and I think I had intended to do that as a separate PR. Sorry for asking you fix that just to re-do it here and have me done it in the same way 🤦

Comment (enov) [reply .github/workflows/test.yml]: No worries :smile:

File: package.json (hunk 1 above)
Comment (enov) [reply package.json:28]: What about [peerDependencies](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies)? I see that npm people flip-flopped on whether or not peer dependencies should be installed by default. As of npm v7 they are installed by default :unamused:. I do not see any other package.json dependency type  where the dependencies are not installed by default, but only suggested to be installed.

Comment (wesleytodd) [reply package.json:28]: Yeah so the problem reported was an install hang: https://github.com/wesleytodd/express-openapi/issues/38

So my thinking was that we need to avoid installing that package entirely.

Comment (wesleytodd) [reply package.json:28]: I guess I didn't test it fully though, so I guess I would need to publish a prerelease and try to install to see which variant would work.

Comment (enov) [reply package.json:28]: Knex has its peer dependencies in `peerDependenciesMeta`, and marks them as optional https://github.com/knex/knex/blob/master/package.json#L81C3 . This does not install the dependencies and emits no warning. There has to be more info in the readme, so that people relying on the UIs know what to do.

Comment (wesleytodd) [reply package.json:28]: Oh yeah, good call. That should for sure be addressed. Are you interested in me adding you on the repo so we can collab?

Comment (wesleytodd) [reply package.json:28]: > Knex has its peer dependencies in peerDependenciesMeta

Yeah, I think there are plenty of ways to address this, but I guess honestly I am nearing the idea of just dropping the redoc dep entirely? Thoughts on that?

Comment (enov) [reply package.json:28]: We do not use redoc. I would appreciate its removal.
You can add me on the repo. Thanks!

Comment (wesleytodd) [reply package.json]: Ok, updated this to remove redoc. Will add you now. I did end up moving it back to a direct dep just since it was easy, but left the logic to make things optional. Thoughts on even just droping this whole PR and instead just having a new one to remove redoc?

Comment (enov) [reply package.json]: Agreed, let's take care of the redoc removal first.

**Comment by wesleytodd** [comment]
Closed in favor of #52
